unity-renderer/Assets/Scripts/MainScripts/DCL/Components/CameraModeArea/CameraModeArea.cs

```diff
@@ -0,0 +1,176 @@
+
+        CameraMode.ModeId ICameraModeArea.cameraMode => areaModel.cameraMode;
+
+        IDCLEntity IEntityComponent.entity => areaEntity;
+
+        IParcelScene IComponent.scene => areaScene;
+
+        string IComponent.componentName => "CameraModeArea";
+
+        void IComponent.UpdateFromJSON(string json)
+        {
+            OnModelUpdated(areaModel.GetDataFromJSON(json) as Model);
+        }
+
+        void IComponent.UpdateFromModel(BaseModel model)
+        {
+            OnModelUpdated(model as Model);
+        }
+
+        IEnumerator IComponent.ApplyChanges(BaseModel model) => null;
+
+        void IComponent.RaiseOnAppliedChanges() { }
+
+        bool IComponent.IsValid() => true;
+
+        BaseModel IComponent.GetModel() => areaModel;
+
+        int IComponent.GetClassId() => (int)CLASS_ID_COMPONENT.CAMERA_MODE_AREA;
+
+        Transform IMonoBehaviour.GetTransform() => areaEntity?.gameObject.transform;
+
+        void ICleanable.Cleanup()
+        {
+            Dispose();
+        }
+
+        void IEntityComponent.Initialize(IParcelScene scene, IDCLEntity entity)
+        {
+            Initialize(scene, entity, Environment.i.platform.updateEventHandler, DataStore.i.player.playerCollider.Get());
+        }
+
+        internal void OnModelUpdated(in Model newModel)
+        {
+            if (!IsValidCameraMode(newModel.cameraMode))
+            {
+                newModel.cameraMode = CommonScriptableObjects.cameraMode.Get();
+            }
+
+            bool cameraModeChanged = newModel.cameraMode != areaModel.cameraMode;
+            areaModel = newModel;
+
+            if (cameraModeChanged && isPlayerInside)
+            {
+                areasController.ChangeAreaMode(this, areaModel.cameraMode);
+            }
+        }
+
+        internal void Initialize(in IParcelScene scene, in IDCLEntity entity, in IUpdateEventHandler updateEventHandler, in Collider playerCollider)
+        {
+            areaScene = scene;
+            areaEntity = entity;
+            this.updateEventHandler = updateEventHandler;
+            this.playerCollider = playerCollider;
+
+            updateEventHandler.AddListener(IUpdateEventHandler.EventType.Update, Update);
+        }
+
+        internal void Update()
+        {
+            bool playerInside = IsPlayerInsideArea();
+
+            switch (playerInside)
+            {
+                case true when !isPlayerInside:
+                    areasController.AddInsideArea(this);
+                    break;
+                case false when isPlayerInside:
+                    areasController.RemoveInsideArea(this);
+                    break;
+            }
+            isPlayerInside = playerInside;
+        }
+
+        internal void Dispose()
+        {
+            OnAreaDisabled();
+            updateEventHandler.RemoveListener(IUpdateEventHandler.EventType.Update, Update);
+        }
+
+        private bool IsValidCameraMode(in CameraMode.ModeId mode)
```

Comment:
How can the camera mode be invalid? I always find bit shifting/comparing a bit hard to read/maintain